sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java

```diff
@@ -120,6 +122,40 @@ private static String nextAssertionName() {
   // Do not instantiate.
   private PAssert() {}
 
+  private static final class DefaultConcludeFn extends DoFn<SuccessOrFailure, Void> {
+
+    private Counter successCounter;
+    private Counter failureCounter;
+
+    @StartBundle
+    public void startBundle(Context c) {
+      successCounter = Metrics.counter(PAssert.class, PAssert.SUCCESS_COUNTER);
```

Comment:
I don't think these need to be defined in startBundle -- Counter should be serializable, so this could be:

`private Counter successCounter = Metrics.counter(PAssert.class, PAssert.SUCCESS_COUNTER);`

In fact, you could even define these in the `PAssert` class itself, rather than just the string names.